lib/rules/no-this-in-template-only-components.js

```diff
@@ -24,6 +24,15 @@ function isTemplateOnlyComponent(hbsFilePath, validComponentExtensions) {
     return validComponentExtensions.some((ext) => fs.existsSync(pathWithoutExtension + ext));
   }
   const paths = path.normalize(hbsFilePath).split(path.sep);
+
+  while (paths.length > 0 && paths[0] !== 'app' && paths[0] !== 'addon') {
```

Comment:
some addons with templates may not have an `addon` directory (v2 addons and v2 apps may use `src`)

Why check for specific path segments?

Reply:
@NullVoxPopuli you absolutely right, but this rule is kinda "happy-path" rule, and works only for classic layouts. In scope of this PR we don't change rule logic, we only adding support for absolute paths (before - only relative works)